### ISSUE TITLE
Suffix of generated types should maybe be "Result" instead of "Query"

### ISSUE THREAD

**Issue by jonaskello** [apollographql/apollo-codegen#315]
Today the types that are generated are named `GetFooQuery`. Becuase of this naming convention, it is quite easy for someone to assume that that type represents the actual query. But in reality it represents the result/reponse that the query will generate. If the types instead were named `GetFooResult` or something similar it might be easier to understand what the type represents. Or perhaps make it possible for the user to decide the suffix (perhaps it already is but I have not found that option).

The generation examples now state:

```
apollo-codegen generate **/*.graphql --schema schema.json --target typescript --output operation-result-types.ts
```

Which is good becuase the file name `operation-result-types.ts` communicates what the generated types represents. If the type names did the same I think it would clear up some confusion.

**Comment by martijnwalraven** [comment]
Yeah, that's fair. On iOS, we generate actual query classes as `GetFooQuery`, and then a nested `Data` struct for the data part of the result. The whole result is represented as `GraphQLResult<Data>`, which has both `data` and `errors`.

We probably want to think about this in the context of https://github.com/apollographql/apollo-codegen/issues/308, which is about generating query classes for TypeScript or Flow as well.

**Comment by alamothe** [comment]
It should probably be `GetFooQueryResult`

`GetFooQuery` definitely collides with the actual query (`gql` or `graphql` definition)

**Comment by alamothe** [comment]
...It would be great to come up with a convention and settle on naming scheme.

For example, `typescript-modern` does not append `Query`/`Mutation` anymore, so it's incompatible with `typescript` 😞

**Comment by jonaskello** [comment]
I think either a suffix of `Response` or `Result` would be appropriate.

**Comment by nilshartmann** [comment]
I also think `Result` is a good idea. A "larger" solution might be adding some configuration options.

**Comment by nilshartmann** [comment]
Any chance this will be implemented? If your interested, I could try to help implementing...

**Comment by nilshartmann** [comment]
ICYI: I've written a little codemod script that renames the generated types after they have been generated by apollo-codegen (as workaround until this problem gets solved in apollo-codegen)

You can find the transform script here: https://gist.github.com/nilshartmann/39e6dc755c91a60c1dc1911180e154b5

**Comment by jardakotesovec** [comment]
As @alamothe mentioned, suffixes disappeared with 0.20 version, which is breaking change and was not mentioned/explained in CHANGELOG.. so I am confused about the intentions behind it. I personally liked the suffixes.. so would at least appreciate explanation for this change.

**Comment by kgtkr** [comment]
```ts
import { myQuery } from "./my-query.gql";
import { myQuery } from "./__generated__/myQuery";
```

Names conflict

**Comment by brabeji** [comment]
@jbaxleyiii since this issue have been closed, is there any official resolution to this problem? The generated name conflicts with my query  `UserDetailManageScreenQuery` component. Why not `UserDetailManageScreenQueryData` in a same way as `UserDetailManageScreenVariables` are suffixed?

**Comment by wolthers** [comment]
Any resolution to this?

**Comment by PowerKiKi** [comment]
@lewisf you contributed #304 a while ago, which I believe was when the suffix `Query` was removed from generated types in Typescript. Could you please explain why it was decided to drop the suffix `Query` ? was the risk of collision considered during that decision ? And was `Result`, instead of `Query`, ever considered ?

**Comment by ScottyFillups** [comment]
Is there any news on this? A suffix option would be great

**Comment by SurokDima** [comment]
I also think it would be nice to have a configuration option to customize "data" and "variables" suffix for generated typescript types.
